pkg/networkservice/common/serialize/executor_func.go

```diff
@@ -0,0 +1,33 @@
+// Copyright (c) 2020 Doc.ai and/or its affiliates.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package serialize
+
+import "github.com/networkservicemesh/sdk/pkg/tools/multiexecutor"
+
+// ExecutorFunc is a serialize.Executor.AsyncExec func type
```

Comment:
I feel we can just create an interface 
```go
type Executor interface {
	AsyncExec(f func()) <-chan struct{}
}
```

Reply:
We cannot implement interface without creating a type, so it will be a new type AND an interface. Why should we create an interface if it can be created on the user side?

Reply:
>Why should we create an interface if it can be created on the user side?

I do not follow. Could you say more?

> We cannot implement interface without creating a type, so it will be a new type AND an interface. 

Note you could also put function as a interface:

```
type Foo interface {
    Bar()
}
type myFunc func()

func (f myFunc) Bar() {
    f()
}
```

Reply:
OK, added `IExecutor` in `serialize/context.go`.